host/transient_task_test.go

```diff
@@ -132,6 +133,164 @@ func (s *integrationSuite) TestTransientWorkflowTaskTimeout() {
 	s.True(workflowComplete)
 }
 
+func (s *integrationSuite) TestTransientWorkflowTaskHistorySize() {
+	id := "integration-transient-workflow-task-history-size-test"
+	wt := "integration-transient-workflow-task-history-size-test-type"
+	tl := "integration-transient-workflow-task-history-size-test-taskqueue"
+	identity := "worker1"
+
+	// Start workflow execution
+	request := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:           uuid.New(),
+		Namespace:           s.namespace,
+		WorkflowId:          id,
+		WorkflowType:        &commonpb.WorkflowType{Name: wt},
+		TaskQueue:           &taskqueuepb.TaskQueue{Name: tl},
+		Input:               nil,
+		WorkflowRunTimeout:  timestamp.DurationPtr(100 * time.Second),
+		WorkflowTaskTimeout: timestamp.DurationPtr(2 * time.Second),
+		Identity:            identity,
+	}
+
+	we, err0 := s.engine.StartWorkflowExecution(NewContext(), request)
+	s.NoError(err0)
+	s.Logger.Info("StartWorkflowExecution", tag.WorkflowRunID(we.RunId))
+
+	workflowExecution := &commonpb.WorkflowExecution{
+		WorkflowId: id,
+		RunId:      we.RunId,
+	}
+
+	// workflow logic
+	stage := 0
+	workflowComplete := false
+	largeValue := make([]byte, 1024*1024)
+	var sawHistorySize int64
+	wtHandler := func(execution *commonpb.WorkflowExecution, wt *commonpb.WorkflowType,
```

Comment:
Like the other transient task tests, this needs to deliberately fail and time out workflow tasks. I suppose in the sdk you can fail a workflow task by panicking? And time out just by sleeping? But then I'm not sure how it would maintain state across them. Maybe a sideeffect?